library/std/src/sys/itron/thread.rs

```diff
@@ -17,12 +17,18 @@ use crate::{
 };
 
 pub struct Thread {
-    inner: ManuallyDrop<Box<ThreadInner>>,
+    p_inner: *mut ThreadInner,
```

Comment:
Would it make more sense to make this a `NonNull<ThreadInner>` to keep the niche?